source/com.microsoft.tfs.core/src/com/microsoft/tfs/core/telemetry/TfsTelemetryHelper.java

```diff
@@ -75,22 +81,42 @@ public synchronized static TelemetryClient getTelemetryClient() {
     }
 
     public static void sendMetric(final String name, final double value) {
+        if (sendingTelemetryDisabled) {
```

Comment:
In this case I want to avoid creating the configuration instance and the client instance to save as much time as possible. The more code I can skip the more time I save. But I didn't notice that method. I may try that method and profile with it disabled to see if that provides enough of a savings.